crates/nu-command/tests/commands/find.rs

```diff
@@ -97,27 +67,96 @@ fn find_with_filepath_search_with_multiple_patterns() {
-                | get name
-                | find arep ami
-                | to json -r
-            "#
-        ));
+            cwd: dirs.test(),
+            "ls | get name | find arep ami | to json -r"
+        );
 
         assert_eq!(actual.out, r#"["amigos.txt","arepas.clu"]"#);
     })
 }
 
 #[test]
 fn find_takes_into_account_linebreaks_in_string() {
-    let actual = nu!(
-    cwd: ".", pipeline(
-    r#"
-        "atest\nanothertest\nnohit\n" | find a | length
-        "#
-    ));
+    let actual = nu!(r#""atest\nanothertest\nnohit\n" | find a | length"#);
 
     assert_eq!(actual.out, "2");
 }
+
+#[test]
+fn find_with_regex_in_table_keeps_row_if_one_column_matches() {
+    Playground::setup("filepath_test_2", |dirs, sandbox| {
+        sandbox.with_files(vec![
+            EmptyFile("amigos.txt"),
+            EmptyFile("arepas.clu"),
+            EmptyFile("los.txt"),
+            EmptyFile("tres.txt"),
+        ]);
+
+        let actual = nu!(
+            cwd: dirs.test(),
+            "ls | find --regex t.t | get name | to json -r"
+        );
+
+        assert_eq!(actual.out, r#"["amigos.txt","los.txt","tres.txt"]"#);
+    })
+}
+
+#[test]
+fn inverted_find_with_regex_in_table_keeps_row_if_none_of_the_columns_matches() {
+    Playground::setup("filepath_test_2", |dirs, sandbox| {
+        sandbox.with_files(vec![
+            EmptyFile("amigos.txt"),
+            EmptyFile("arepas.clu"),
+            EmptyFile("los.txt"),
+            EmptyFile("tres.txt"),
+        ]);
+
+        let actual = nu!(
+            cwd: dirs.test(),
+            "ls | find --regex t.t --invert | get name | to json -r"
+        );
+
+        assert_eq!(actual.out, r#"["arepas.clu"]"#);
+    })
+}
+
+#[test]
+fn find_in_table_only_keep_rows_with_matches_on_selected_columns() {
+    Playground::setup("filepath_test_2", |dirs, sandbox| {
+        sandbox.with_files(vec![
+            EmptyFile("file.txt"),
+            EmptyFile("arepas.clu"),
+            EmptyFile("los.txt"),
+            EmptyFile("tres.txt"),
+        ]);
+
+        let actual = nu!(
+            cwd: dirs.test(),
+            "ls | find file --columns [name] | get name | to json -r"
+        );
+
+        assert!(actual.out.contains("file"));
+        assert!(!actual.out.contains("arepas"));
+        assert!(!actual.out.contains("los"));
+        assert!(!actual.out.contains("tres"));
+    })
+}
+
+#[test]
+fn inverted_find_in_table_keeps_row_if_none_of_the_selected_columns_matches() {
+    Playground::setup("filepath_test_2", |dirs, sandbox| {
+        sandbox.with_files(vec![
+            EmptyFile("file.txt"),
+            EmptyFile("arepas.clu"),
+            EmptyFile("los.txt"),
+            EmptyFile("tres.txt"),
+        ]);
+
+        let actual = nu!(
+            cwd: dirs.test(),
+            "ls | find file --columns [name] --invert | get name | to json -r"
+        );
+
+        assert_eq!(actual.out, r#"["arepas.clu","los.txt","tres.txt"]"#);
+    })
+}
```

Comment:
i have trouble wrapping my head around these tests, namely to reproduce them on my machine...

could we simplify them and use `[moe larry bob] | wrap names` as in the tests above?
i have the issue on latest `main` where `[moe larry bob] | wrap names | find m -v` returns a table with emtpy cells.
i think these simpler tests would 
- show the same good behaviour
- be simpler to read and understand

:yum:

Reply:
Ohh, I didn't know the wrap command! Yes, tests will be way simpler with that

Reply:
> I have the issue on latest main where [moe larry bob] | wrap names | find m -v returns a table with emtpy cells.
Couldn't reproduce this though. What do you mean by "latest main"?

Reply:
'twas the latest `git` revision of Nushell at the time :wink: